content/information-distribution/node.md

```diff
@@ -54,14 +57,65 @@ the Matrix room to join.
-nodes from this list.
+for this Matrix room. This can be done by processing all of the `m.room.aliases`
+events in the state of the Matrix room, each of them containing the list of
+aliases defined for a given node. The content for such an event looks like this:
+
+```
+"aliases": [
+    "#alias1:example.com",
+    "#alias2:example.com"
+]
+```
+
+The example above shows the content of the `m.room.aliases` state event for the
+node which uses `example.com` as its server name.
+
+Client implementations **must** process these lists of aliases to generate a map
+mapping all of the entry nodes for this Matrix room with the aliases they
+provide for this room.
+
+### Reaching an entry node
+
+Because the list of aliases only list the nodes' server names, and not their
+FQDN, a node might not be reachable at the address  In order to find the
+effective FQDN and port to reach a node at, client implementations **must**
+perform a DNS query on an `SRV` record for the `_matrix._tcp` sub-domain of the
+server's name which, in the example above, would look like
+`_matrix._tcp.example.com`. If such a record doesn't exist, client
+implementations **shoud** consider that the node is reachable using its server
+name as its FQDN, and the default Matrix encrypted federation port, `8448`, as
+its port.
 
 {{% notice note %}}
-Nodes that aren't entry nodes are named "relay nodes".
+Because most Matrix homeservers can also be reached through the standard `443`
+HTTPS port, client implementations **should** try reaching a node using this
+port instead of the `8448` one, and fall back to the latter port (or any port
+defined in the node's `SRV` DNS record) if closed. This would result in making
+the network traffic generated by Informo client implementations more difficult
+to detect and filter on.
 {{% /notice %}}
+
+{{% notice tip %}}
+Web-based client implementations **can** use the [DNS over
+HTTPS](https://tools.ietf.org/html/rfc8484) standard to perform DNS queries
+instead of standard DNS queries which they might not be able to perform.
+{{% /notice %}}
+
+In the event of an entry node that's currently being used becoming unreachable,
+clients implementations **must** use the generated map to provide its user(s)
+with alternative entry nodes and aliases to use. Client implementations **can**
+use received pings in order to filter out the inactive nodes from this list.
+
+### Joining a federation
+
+Once both an entry node and a room alias have been selected as the one to use in
+order to join a federation, whether this selection originated from an explicit
+action by a user or, either partially or entirely, from a client
+implementation's inner mechanisms, client implementations **must** send a [join
+request](https://matrix.org/speculator/spec/HEAD/client_server/unstable.html#post-matrix-client-r0-join-roomidoralias)
+to the selected node using the selected alias.
+
+## Other types of nodes
+
+Nodes that aren't entry nodes are named "relay nodes".
```

Comment:
I suppose these relay nodes don't serve a client-to-server API? What are their purpose / what are the benefits to not serve a C2S API?

Reply:
Act as a relay. If `A` can't reach `B` but can reach `C`, even if `C` isn't an entry node, `A`'s information will eventually manage to reach `B` via `C` (thanks to pings).

Reply:
I added the explanation above to the proposal in https://github.com/Informo/specs/pull/2/commits/9c1df0812969e9c9c2826432f2878cd0a0660b7e.

Reply:
My question was more: why a homeserver administrator would like to disable the C2S API?
What's the point of relay nodes when standard nodes can offer additional services with, afaik no drawbacks.

Reply:
> why a homeserver administrator would like to disable the C2S API?

Seems like there's some kind of confusion here. As defined higher in the page:

> An entry node must allow guest access, and must define at least one alias to the Matrix room

A relay node is just a node that doesn't follow both rules.

Regarding the reasons that could lead a HS admin to use their HS as a relay node rather than an entry node, a few examples would be that they don't want the exposure of an entry node, or they don't want to handle the load that's put on the C-S API by operating an entry node, or they want to be more restrictive about the access to the HS's C-S API, etc., but still want to contribute to Informo's network.

Reply:
ok, sounds legit.
I got confused thinking a relay node was a Synapse without any C2S API, but there are indeed other requirements to be an entry node.

I would change it to:

> Nodes that aren't entry nodes (like a node without guest access) are named "relay nodes".

just to make clear that many homeservers can be unsuited for being an entry node.